core/domain/suggestion_registry_test.py

```diff
@@ -759,8 +759,9 @@ def test_validate_change_question_state_data_schema_version(self):
 
         suggestion.validate()
 
-        suggestion.change.question_dict[
-            'question_state_data_schema_version'] = 0
+        question_dict = suggestion.change.question_dict
```

Comment:
Could you explain why you are doing this change? Ditto below.

Reply:
pylint errors here if we changed suggestion.change.question_dict property directly:

```
************* Module oppia.core.domain.suggestion_registry_test
E:762, 8: 'suggestion.change.question_dict' does not support item assignment (unsupported-assignment-operation)
E:812, 8: 'suggestion.change.question_dict' does not support item assignment (unsupported-assignment-operation)
```

Reply:
Was it initialized to None? http://pylint-messages.wikidot.com/messages:e1137

Reply:
No, it isn't. I have just changed to `QuestionChange` object to inherit from `change_domain.BaseChange` after which this error occurs.

Reply:
This needs a bit more digging, too.

Reply:
I tried finding a reason behind this but I am still unable to do that. There is something weird happening in the test:

The first test [here](https://github.com/oppia/oppia/blob/develop/core/domain/suggestion_registry_test.py#L730) is setting the `question_dict ` to None but that should not affect the other tests since the test functions are independent. Strangely, if I remove this none setting pylint does not give any error.

The two test where error is produced are on [line 749](https://github.com/oppia/oppia/blob/develop/core/domain/suggestion_registry_test.py#L749) and [line 794]
(https://github.com/oppia/oppia/blob/develop/core/domain/suggestion_registry_test.py#L794). Now if do not remove the None assignment, pylint gives error for both these tests. Further if I keep the test on line 794 as it is and the None assignment also but change the test on line 749, as have done in the PR, there is no error for test on line 794.

This thing does not happen on develop. The changes are made to `QuestionChange` class which now inherits from `BaseChange` class. In `init` the attributes are set as follows:

```
for attribute_name in cmd_attribute_names:
            setattr(self, attribute_name, change_dict.get(attribute_name))
```

Earlier the attributes were set as:
```
        if self.cmd == CMD_CREATE_NEW_FULLY_SPECIFIED_QUESTION:
            self.question_dict = change_dict['question_dict']
```

If I add this to `init` in `BaseChange` class, the pylint error is gone. I am unable to find a reason of why that is happening.

Further, I ran the test and raised an Exception to print the question_dict in both the tests on line 749 and line 794 and a proper question_dict is produced as output. 

I then checked the pylint code for this which is in the function [here](https://github.com/PyCQA/pylint/blob/master/pylint/checkers/utils.py#L935). I edited the function locally by adding `raise Exception(value)` after [this line](https://github.com/PyCQA/pylint/blob/master/pylint/checkers/utils.py#L938). The value is shown as None but in running the backend test the value is not None. I am not sure why pylint is producing a false positive here. There is one issue which I found where pylint produces a false positive for numpy: [Ref](https://github.com/PyCQA/pylint/issues/2767). I am not sure if this issue is related to our case or not. Can you please guide me here as what should I do next?

\cc @apb7 @lilithxxx

Reply:
Let's leave it, but put a comment in the code that links to this (excellent) analysis, and file an issue to investigate further and figure out what the fix is. It might be worth filing an issue on the pylint repo as well, since it's possible that this is an error on their end. Thank you for following up on this!

Reply:
You're welcome and thank you for asking me to dig further into this instead of leaving it as it is! I have filed an issue #7008 as well as added a comment in the code.